examples/gpu-printing/main.cpp

```diff
@@ -113,9 +113,13 @@ Result execute()
     printBufferViewDesc.type = IResourceView::Type::UnorderedAccess;
     auto printBufferView = gDevice->createBufferView(printBuffer, printBufferViewDesc);
 
+    ITransientResourceHeap::Desc transientResourceHeapDesc = {};
+    transientResourceHeapDesc.constantBufferSize = 256;
+    auto transientHeap = gDevice->createTransientResourceHeap(transientResourceHeapDesc);
+
     ICommandQueue::Desc queueDesc = {ICommandQueue::QueueType::Graphics};
     auto queue = gDevice->createCommandQueue(queueDesc);
-    auto commandBuffer = queue->createCommandBuffer();
+    auto commandBuffer = transientHeap->createCommandBuffer();
```

Comment:
Minor: when we get around to supporting Metal, we will need to deal with command buffers being allocated from a specific queue, so this either needs to be `queue->createCommandBuffer(heap)` or `heap->createCommandBuffer(queue)` to make sure that both objects are available.

I lean toward `queue->createCommandBufer(heap)` because in many cases we can/should support allocations with a null heap, which would be used to indicate that the user want a persistent allocation.